### PR TITLE
Functional Tests : `functional/BO/14*/12*/*` : Changes for Scopes

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : `functional/BO/14*/12*/*` : Changes for Scopes
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: 
| UI Tests          | https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/6786245635
| Fixed issue or discussion?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

Issue Found : #34496

### CODE REVIEW

**Review by nicosomb — DISMISSED** [review]

**Review by tleon — DISMISSED** [review]

File: tests/UI/campaigns/modules/ps_apiresources/01_installation/01_uninstallAndInstallModule.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+      const pageTitle = await addNewApiAccessPage.getPageTitle(page);
+      expect(pageTitle).to.eq(addNewApiAccessPage.pageTitleCreate);
+    });
+
+    it('should check that scopes from Core are present and enabled', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkScopesCore', baseContext);
+
+      const scopes = await addNewApiAccessPage.getApiScopes(page, '__core_scopes');
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/modules/ps_apiresources/01_installation/01_uninstallAndInstallModule.ts]:
```suggestion
```

File: tests/UI/campaigns/modules/ps_apiresources/01_installation/01_uninstallAndInstallModule.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      expect(scopes.length).to.be.gt(0);
+    });
+
+    it('should check that scopes from Module are not present', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkScopeModule', baseContext);
+
+      const scopes = await addNewApiAccessPage.getApiScopes(page, Modules.psApiResources.tag);
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/modules/ps_apiresources/01_installation/01_uninstallAndInstallModule.ts]:
```suggestion
```

File: tests/UI/campaigns/modules/ps_apiresources/01_installation/01_uninstallAndInstallModule.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+      const pageTitle = await addNewApiAccessPage.getPageTitle(page);
+      expect(pageTitle).to.eq(addNewApiAccessPage.pageTitleCreate);
+    });
+
+    it('should check that scopes from Core are present and enabled', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkScopesCore', baseContext);
+
+      const scopes = await addNewApiAccessPage.getApiScopes(page, '__core_scopes');
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/modules/ps_apiresources/01_installation/01_uninstallAndInstallModule.ts]:
```suggestion
```

File: tests/UI/campaigns/modules/ps_apiresources/01_installation/01_uninstallAndInstallModule.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+      expect(scopes.length).to.be.gt(0);
+    });
+
+    it('should check that scopes from Module are not present', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkScopeModule', baseContext);
+
+      const scopes = await addNewApiAccessPage.getApiScopes(page, Modules.psApiResources.tag);
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/modules/ps_apiresources/01_installation/01_uninstallAndInstallModule.ts]:
```suggestion
```

**Review by matthieu-rolland — DISMISSED** [review]

File: tests/UI/campaigns/modules/ps_apiresources/01_installation/02_disableEnableModule.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+// Import commonTests
+import loginCommon from '@commonTests/BO/loginBO';
+
+// Import pages
+// Import BO pages
+import apiAccessPage from '@pages/BO/advancedParameters/APIAccess';
+import addNewApiAccessPage from '@pages/BO/advancedParameters/APIAccess/add';
+import featureFlagPage from '@pages/BO/advancedParameters/featureFlag';
+import setFeatureFlag from '@commonTests/BO/advancedParameters/newFeatures';
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/modules/ps_apiresources/01_installation/02_disableEnableModule.ts]:
```suggestion
```

File: tests/UI/campaigns/modules/ps_apiresources/01_installation/02_disableEnableModule.ts [hunk 6]
```diff
@@ -0,0 +1,156 @@
+// Import utils
+import basicHelper from '@utils/basicHelper';
+import helper from '@utils/helpers';
+import testContext from '@utils/testContext';
+
+// Import commonTests
+import loginCommon from '@commonTests/BO/loginBO';
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/modules/ps_apiresources/01_installation/02_disableEnableModule.ts:8]:
```suggestion
import loginCommon from '@commonTests/BO/loginBO';
import setFeatureFlag from '@commonTests/BO/advancedParameters/newFeatures';
```

File: tests/UI/campaigns/modules/ps_apiresources/01_installation/02_disableEnableModule.ts [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+        const pageTitle = await addNewApiAccessPage.getPageTitle(page);
+        expect(pageTitle).to.eq(addNewApiAccessPage.pageTitleCreate);
+      });
+
+      it('should check that scopes from Core are present and enabled', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `checkScopesCore${index}`, baseContext);
+
+        const scopes = await addNewApiAccessPage.getApiScopes(page, '__core_scopes');
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/modules/ps_apiresources/01_installation/02_disableEnableModule.ts]:
```suggestion
```

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by tleon — DISMISSED** [review]

File: tests/UI/pages/BO/advancedParameters/APIAccess/add.ts [hunk 8]
```diff
@@ -83,12 +89,84 @@ class AddNewAPIAccess extends BOBasePage {
...
+    // eslint-disable-next-line no-restricted-syntax
+    for (const scope of await page.locator(`#api_access_scopes_${group}_accordion div.switch-scope`).all()) {
+      const dataScope: string|null = await scope.getAttribute('data-scope');
+
+      if (dataScope) {
+        scopes.push(dataScope);
+      }
+    }
+
+    return scopes;
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/advancedParameters/APIAccess/add.ts]: 1. Maybe declare this selector like this `this.accessScopBlock = (group) => `#api_access_scopes_${group}_accordion div.switch-scope`;

2. Why not use `evaluateAll` like this : 
```js
scops = await page.locator(this.accessScopBlock)
  .evaluateAll(
    (all) => all
      .map(
        (el) => el.getAttribute('data-scope')
      )
      .filter(
        (attr) => attr != undefined
      )
    );
```

File: tests/UI/pages/BO/advancedParameters/APIAccess/add.ts [hunk 9]
```diff
@@ -83,12 +89,84 @@ class AddNewAPIAccess extends BOBasePage {
...
+  /**
+   * Check/Uncheck a specific API Scope
+   * @param page {Page} Browser tab
+   * @param scope {string} Scope
+   * @param valueWanted {boolean} True if we need to enable status, false if not
+   * @return {Promise<boolean>} return true if action is done, false otherwise
+   */
+  async setAPIScopeChecked(page: Page, scope: string, valueWanted: boolean = true): Promise<boolean> {
+    if (await this.isAPIScopeChecked(page, scope) !== valueWanted) {
+      await page.click(this.scopeStatus(scope));
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/advancedParameters/APIAccess/add.ts]:
```js
await page.locator(this.scopeStatus(scope)).click();
```

Comment (Progi1984) [reply tests/UI/pages/BO/advancedParameters/APIAccess/add.ts]: Fixed

File: tests/UI/pages/BO/advancedParameters/APIAccess/add.ts (hunk 8 above)
Comment (Progi1984) [reply tests/UI/pages/BO/advancedParameters/APIAccess/add.ts]: I check if it's ok.

Comment (Progi1984) [reply tests/UI/pages/BO/advancedParameters/APIAccess/add.ts]: Fixed

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by tleon — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @Progi1984

**Pull request merged**